pkg/api/query.go

```diff
@@ -14,12 +14,12 @@ import (
 	"github.com/timescale/promscale/pkg/promql"
 )
 
-func Query(conf *Config, queryEngine *promql.Engine, queryable promql.Queryable) http.Handler {
-	hf := corsWrapper(conf, queryHandler(queryEngine, queryable))
+func Query(conf *Config, queryEngine *promql.Engine, queryable promql.Queryable, metrics *Metrics) http.Handler {
+	hf := corsWrapper(conf, queryHandler(queryEngine, queryable, metrics))
 	return gziphandler.GzipHandler(hf)
 }
 
-func queryHandler(queryEngine *promql.Engine, queryable promql.Queryable) http.HandlerFunc {
+func queryHandler(queryEngine *promql.Engine, queryable promql.Queryable, metrics *Metrics) http.HandlerFunc {
```

Comment:
This is good. But I think we need to add two metrics: the equivalent of QueryBatchDuration (call it QueryDuration) that times the queries. And the equivalent of InvalidReadReqs (call it InvalidQueryReqs) which traces the queries with incorrect parameters.